src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -803,9 +806,20 @@ public IMethodSymbol GetEntryPoint(CancellationToken cancellationToken)
         /// Get the symbol for the predefined type from the Cor Library referenced by this
         /// compilation.
         /// </summary>
-        public INamedTypeSymbol GetSpecialType(SpecialType specialType)
+        public INamedTypeSymbol? GetSpecialType(SpecialType specialType)
+        {
+            return (INamedTypeSymbol?)CommonGetSpecialType(specialType)?.GetITypeSymbol();
+        }
+
+        public INamedTypeSymbol GetSpecialTypeOrThrow(SpecialType specialType)
         {
-            return (INamedTypeSymbol)CommonGetSpecialType(specialType)?.GetITypeSymbol();
+            var symbol = GetSpecialType(specialType);
+            if (symbol is null)
+            {
+                throw new InvalidOperationException($"Special type {specialType} is not available.");
```

Comment:
```suggestion
                throw new InvalidOperationException($"Special type '{specialType}' is not available.");
```

Reply:
💭 I'd probably just write it like this:

```
return GetSpecialType(specialType) ?? throw ...
```

But I can see potential value in both ways.